api/router.go

```diff
@@ -0,0 +1,58 @@
+package api
+
+import (
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/numary/ledger/api/controllers"
+	"github.com/numary/ledger/api/middlewares"
+	"github.com/numary/ledger/ledger"
+)
+
+// Router -
+func Router(cc cors.Config, resolver *ledger.Resolver) *gin.Engine {
+	router := gin.Default()
+
+	// Default Middlewares
+	router.Use(
+		cors.New(cc),
+		gin.Recovery(),
+		middlewares.AuthMiddleware(router),
+	)
+
+	// API Controllers
+	configController := controllers.CreateConfigController()
+	ledgerController := controllers.CreateLedgerController()
+	transactionController := controllers.CreateTransactionController()
+	accountController := controllers.CreateAccountController()
+	scriptController := controllers.CreateScriptController()
```

Comment:
I think I'd be inclined to have the controllers as well created and injected via fx. What do you think?

Reply:
Never tried fx before. I changed it, but I think it can be improved still. What do you think?